contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,36 +130,226 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
         tempvar pedersen_ptr = pedersen_ptr;
     }
 
-    // Capture back if held by loser
-    // Check if loser has winners token
-    let (winners_relic_owner) = get_current_relic_holder(winner_token_id);
+    let (realms_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Realms
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
-        return ();
+    let (remove_relics_len) = loop_claim_order_relic(
+        0, realms_address, winner_realm_data, loser_token_id, loser_relics_len, 0, remove_relics
+    );
+
+    loop_remove_relic(0, 0, loser_relics_len, old_owner_token_id, remove_relics_len, remove_relics);
+    owned_relics_len.write(old_owner_token_id, loser_relics_len - remove_relics_len);
```

Comment:
Have you tested this? Does this work as intended? What if the first branch above matches (line 112 and below)? Then `loser_relics_len` is already off by one (line 114), no?

Furthermore, `loop_claim_order_relic` calls `_set_relic_holder` which sets the `owned_relics_len` value. Then `loop_remove_relic` updates the `owned_relics` array without updating the `_len` (IMO this should be done in lockstep). And finally, the `owned_relics_len` is once more updated here.